(PA-4787) Warnings when running embedded ruby

chutzimir opened the pull request:
Getting warnings when running the wrapped binaries when bundler has
exported some environment variables. The fix is to clean out the
environment before running the ruby process which strips the
`LD_LIBRARY_PATH`.

puppetlabs-jenkins commented:
Can one of the admins verify this patch?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/puppetlabs/puppet-agent?pullRequest=2268) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/puppetlabs/puppet-agent?pullRequest=2268) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/puppetlabs/puppet-agent?pullRequest=2268) it.</sub>

chutzimir commented:
I cannot make a Jira ticket as Jira is not letting me make an account (it says my email is already in use), and if I request username recovery, I don't get any email. So I'm a bit stuck on how to report this issue.

joshcooper commented:
Thanks for submitting this PR @chutzimir. Let me check with some folks to see about the JIRA issue.

chutzimir commented:
To clarify how to reproduce the issue, I use the `rt` binary from a subshell

chutzimir commented:
> Thanks for submitting this PR @chutzimir. Let me check with some folks to see about the JIRA issue.

I was able to get in Jira and made a ticket. Also updated the commit to have the ticket number and rebased the branch on the latest main.

joshcooper commented:
What is the `rt` cli? Can you provide steps to reproduce?

chutzimir commented:
> What is the `rt` cli? Can you provide steps to reproduce?

The full details are in https://puppet.atlassian.net/browse/PA-4787

The `rt` may have been a mistake on my end. I reproduced this with shelling a call to `facter`. Though I have not been working in this environment for a while now.

AriaXLi commented:
Hello, @chutzimir thank you for your contribution! Since this repository is being archived, this PR will be closed. To see more information on this, see https://www.puppet.com/blog/open-source-puppet-updates-2025.
